backbone.js

```diff
@@ -543,6 +543,18 @@
       return this.set(attr, void 0, _.extend({}, options, {unset: true}));
     },
 
+    // accepts an attribute name (or array of attribute names) and a transformer
+    // function which maps the current attribute value(s) to new attribute value(s)
+    update: function(attrName, transformer, setOptions) {
+      var attrNames = _.isArray(attrName) ? attrName : [attrName];
+      var newValues = _.reduce(attrNames, function(valueCollector, attr) {
+        valueCollector[attr] = transformer.call(this, this.get(attr));
```

Comment:
I think it would be a good idea to pass the model into the transformer as a 3rd argument.  That allows you to do transforms that rely on other attributes.  That probably won't be the greatest code, but it is something that people will probably want to do.